src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/actionfilter/FieldCapsFilter.kt

```diff
@@ -0,0 +1,308 @@
+            val type = fieldMapping.sourceType!!
+            if (response[fieldName] == null) {
+                response[fieldName] = mutableMapOf()
+            }
+            val isSearchable = fieldMapping.fieldType == RollupFieldMapping.Companion.FieldType.DIMENSION
+            response[fieldName]!![type] = FieldCapabilities(fieldName, type, isSearchable, true, fieldMappingIndexMap.getValue(fieldMapping)
+                    .toTypedArray(), null, null, mapOf<String, Set<String>>())
+        }
+
+        return response
+    }
+
+    private fun populateSourceFieldMappingsForRollupJob(rollup: Rollup): Set<RollupFieldMapping> {
+        val rollupFieldMappings = rollup.populateFieldMappings()
+        val sourceIndices = indexNameExpressionResolver.concreteIndexNames(clusterService.state(), IndicesOptions.lenientExpand(), rollup.sourceIndex)
+        sourceIndices.forEach {
+            val mappings = clusterService.state().metadata.index(it).mapping()?.sourceAsMap ?: return rollupFieldMappings
+            rollupFieldMappings.forEach { fieldMapping ->
+                val fieldType = getFieldType(fieldMapping.fieldName, mappings)
+                if (fieldType != null) {
+                    fieldMapping.sourceType(fieldType)
+                }
+            }
+        }
+
+        return rollupFieldMappings
+    }
+
+    private fun populateSourceFieldMappingsForRollupIndex(rollupIndex: String): Map<String, Set<RollupFieldMapping>> {
+        val fieldMappings = mutableMapOf<String, MutableSet<RollupFieldMapping>>()
+        val rollupJobs = clusterService.state().metadata.index(rollupIndex).getRollupJobs() ?: return fieldMappings
+        rollupJobs.forEach { rollup ->
+            if (fieldMappings[rollup.targetIndex] == null) {
+                fieldMappings[rollup.targetIndex] = mutableSetOf()
+            }
+            fieldMappings[rollup.targetIndex]!!.addAll(populateSourceFieldMappingsForRollupJob(rollup))
+        }
+        return fieldMappings
+    }
+
+    @Suppress("NestedBlockDepth")
+    private fun populateSourceFieldMappingsForRollupIndices(rollupIndices: Set<String>): MutableMap<RollupFieldMapping, MutableSet<String>> {
+        val fieldMappingsMap = mutableMapOf<RollupFieldMapping, MutableSet<String>>()
+
+        rollupIndices.forEach { rollupIndex ->
+            val fieldMappings = populateSourceFieldMappingsForRollupIndex(rollupIndex)
+            fieldMappings.forEach { rollupIndexFieldMappings ->
+                rollupIndexFieldMappings.value.forEach { fieldMapping ->
+                    if (fieldMappingsMap[fieldMapping] == null) {
+                        fieldMappingsMap[fieldMapping] = mutableSetOf()
+                    }
+                    fieldMappingsMap[fieldMapping]!!.add(rollupIndexFieldMappings.key)
+                }
+            }
+        }
+
+        return fieldMappingsMap
+    }
+
+    private fun getFieldType(fieldName: String, mappings: Map<*, *>): String? {
```

Comment:
This seems very familiar to that other function we have somewhere in.. mappings service? We can extract that out into a common util method to reuse in both places?

Reply:
makes sense, will update

Reply:
moved common code to ``IndexUtils.getFieldFromMappings``